src/Dotnet.Script.DependencyModel/ProjectSystem/ScriptProjectProvider.cs

```diff
@@ -25,7 +26,7 @@ public string CreateProjectForRepl(string code, string targetDirectory, string d
         {
             var parseresult = _scriptParser.ParseFromCode(code);
 
-            targetDirectory = Path.Combine(targetDirectory, "REPL");
+            targetDirectory = Path.Combine(targetDirectory, "interactive", Guid.NewGuid().ToString());
```

Comment:
Would this mean that we get a different temp folder every time it is executed? If so, would the temp folder pile up with csproj files ? 😄

Reply:
so each time we need to resolve a a REPL #load/#r or invoke an eval, we should give it a unique context - otherwise we can have collisions (which is why tests were failing). For example, at the moment having two simultaneous REPL sessions from the same folder potentially creates all kinds of problems. The same applies to eval - if you issue 5 simultaneous evals they'd conflict with each other
I agree it's not perfect but I don't have a better idea - unless we say we allow 1 REPL session per folder and 1 concurrent eval per folder. This way we can drop the guid and just use `repl`/`eval` markers. Maybe this would be less invasive for now?

Reply:
I think that would be better. On windows these temp folder are not cleared upon restart and we could easily end up with a ton of these temp folders. On *nix they are cleared, but *nix machines are "never" restarted 😄